lib/media/track/remotevideotrack.js

```diff
@@ -57,6 +58,17 @@ class RemoteVideoTrack extends RemoteMediaVideoTrack {
       _invisibleElements: {
         value: new WeakSet(),
       },
+      _resizeObserver: {
+        value: new options.ResizeObserver(entries => {
+          // NOTE(mpatwardhan): we ignore elements in _invisibleElements
+          // to ensure that ResizeObserver does not end-up turning off a track when a fresh Video element is
+          // attached and IntersectionObserver has not had its callback executed yet.
+          const visibleElementResized = entries.find(entry => !this._invisibleElements.has(entry.target));
```

Comment:
@makarandp0 Is it possible for the resize observer to fire events for invisible elements? I don't think invisible elements can be resized, no? Unless you are talking about video elements without fixed widths and heights.

Reply:
@manjeshbhargav, Yes, ResizeObserver does get fire its callback even when the element is not visible. I found that by changing dimensions of the elements were not visible in the scrollView. 
This check is most obvious  for freshly added elements for which resizeObserver gets called before interSectionObserver.